apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/ClientPool.cs

```diff
@@ -24,46 +27,60 @@
 
 namespace Google.Cloud.Spanner.Data
 {
+
     internal static class ClientPool
     {
-        private static readonly ConcurrentDictionary<ClientPoolKey, ClientPoolEntry> s_clientEntryPool =
-            new ConcurrentDictionary<ClientPoolKey, ClientPoolEntry>();
+        private static readonly ConcurrentDictionary<ClientCredentialKey, CredentialClientPool> s_clientPoolByCredential =
+            new ConcurrentDictionary<ClientCredentialKey, CredentialClientPool>();
 
         public static async Task<SpannerClient> AcquireClientAsync(
             ITokenAccess credentials = null,
             ServiceEndpoint endpoint = null)
         {
-            var key = new ClientPoolKey(credentials, endpoint ?? SpannerClient.DefaultEndpoint);
-            var poolEntry = s_clientEntryPool.GetOrAdd(key, k => new ClientPoolEntry(key));
-            var result = await poolEntry.AcquireClientFromEntryAsync().ConfigureAwait(false);
-            Logger.LogPerformanceCounter("SpannerClient.Count", () => s_clientEntryPool.Count);
+            var key = new ClientCredentialKey(credentials, endpoint ?? SpannerClient.DefaultEndpoint);
+            var poolEntry = s_clientPoolByCredential.GetOrAdd(key, k => new CredentialClientPool(key));
```

Comment:
yeah they are identical.  I think the risk here is pretty negligible, but I dont mind switching to k.